src/webui/www/private/scripts/fs.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+var QB_EXT = '.!qB';
+
+var pathSeparator = '/';
+
+new Request({
+    url: 'api/v2/app/pathSeparator',
+    method: 'get',
+    onSuccess: function(response) {
+        pathSeparator = response;
+    }
+}).send();
+
+// This file is the JavaScript implementation of base/utils/fs.cpp
+
+function fromNativePath(path) {
+    if (pathSeparator === '/')
+        return path;
+    return path.replace(/\\/g, '/');
+}
+
+/**
+ * Returns the file extension part of a file name.
+ */
+function fileExtension(filename) {
+    var name = filename.replace(QB_EXT, '');
+    var pointIndex = name.lastIndexOf('.');
+    if (pointIndex >= 0)
```

Comment:
I would prefer to write `(pointIndex === -1)` to have the same style with the next 2 functions.